nifi-nar-bundles/nifi-oauth-bundle/nifi-oauth/src/main/java/org/apache/nifi/oauth/httpclient/OAuthHTTPConnectionClient.java

```diff
@@ -0,0 +1,252 @@
+            if (c instanceof HttpURLConnection) {
+                HttpURLConnection httpURLConnection = (HttpURLConnection) c;
+
+                if (headers != null && !headers.isEmpty()) {
+                    for (Map.Entry<String, String> header : headers.entrySet()) {
+                        httpURLConnection.addRequestProperty(header.getKey(), header.getValue());
+                    }
+                }
+
+                if (request.getHeaders() != null) {
+                    for (Map.Entry<String, String> header : request.getHeaders().entrySet()) {
+                        httpURLConnection.addRequestProperty(header.getKey(), header.getValue());
+                    }
+                }
+
+                if (OAuthUtils.isEmpty(requestMethod)) {
+                    httpURLConnection.setRequestMethod(OAuth.HttpMethod.GET);
+                } else {
+                    httpURLConnection.setRequestMethod(requestMethod);
+                    setRequestBody(request, requestMethod, httpURLConnection);
+                }
+
+                httpURLConnection.connect();
+
+                InputStream inputStream;
+                responseCode = httpURLConnection.getResponseCode();
+                if (responseCode == 400 || responseCode == 405 || responseCode == 401 || responseCode == 403) {
+                    inputStream = httpURLConnection.getErrorStream();
+                } else {
+                    inputStream = httpURLConnection.getInputStream();
+                }
+
+                responseHeaders = httpURLConnection.getHeaderFields();
+                responseBody = inputStream;
+            }
+        } catch (IOException e) {
+            throw new OAuthSystemException(e);
+        }
+
+        CustomOAuthAccessTokenResponse cr = new CustomOAuthAccessTokenResponse(responseBody, c.getContentType(), responseCode, responseHeaders,
+                accessTokenName, tokenTypeName, scopeName, expireInName, expireTimeName);
+
+        return (T) cr;
+    }
+
+    private void setRequestBody(OAuthClientRequest request, String requestMethod, HttpURLConnection httpURLConnection)
+            throws IOException {
+        String requestBody = request.getBody();
+        if (OAuthUtils.isEmpty(requestBody)) {
+            return;
+        }
+
+        if (OAuth.HttpMethod.POST.equals(requestMethod) || OAuth.HttpMethod.PUT.equals(requestMethod)) {
+            httpURLConnection.setDoOutput(true);
+            OutputStream ost = httpURLConnection.getOutputStream();
+            PrintWriter pw = new PrintWriter(ost);
+            pw.print(requestBody);
+            pw.flush();
+            pw.close();
+        }
+    }
+
+    @Override
+    public void shutdown() {
+        // Nothing to do here
+    }
+
+
+    public static class CustomOAuthAccessTokenResponse
+        extends OAuthAccessTokenResponse {
+
+        // Names of the fields that should be pulled from the JSON response.
+        private String accessTokenName = null;
```

Comment:
It appears the following fields are unused in this class: `accessTokenName, tokenTypeName, scopeName, expireInName, expireTimeName, responseCode, expireTime`.

Reply:
These fields are used. They are used to determine which key to use when retrieving the OAuth values from the resulting OAuth JSON response. In other words the OAuth2 spec does not require that "access_token" be that exact key for example. It could be something like "AccessToken" This allows for the users to configure what key should be used when pulling from that JSON response. These fields are populated by the user by filling in PropertyDescriptors in AbstractOAuthControllerService.java

Reply:
These fields are used. They are used to determine which key to use when retrieving the OAuth values from the resulting OAuth JSON response. In other words the OAuth2 spec does not require that "access_token" be that exact key for example. It could be something like "AccessToken" This allows for the users to configure what key should be used when pulling from that JSON response. These fields are populated by the user by filling in PropertyDescriptors in AbstractOAuthControllerService.java

Reply:
I get your reasoning, I'm saying that these specific fields in this class aren't used. In the constructor, you set the value of each of these first five fields to the incoming parameter on lines 165 - 169, but then the values of the fields are never accessed (they are `private`, there are no getters, and they are not read within this class). Same for `responseCode` and `expiresIn` (in fact, you call `setResponseCode()` and then explicitly assign `this.responseCode` in consecutive lines?).

Reply:
ahhh ... this is a great reason why you shouldn't try to look at this stuff while running through the airport because you miss stuff like this =)